.github/workflows/proto.yml

```diff
@@ -1,13 +1,30 @@
-name: buf
+name: Proto
 on:
   pull_request:
   push:
     branches:
         - main
 jobs:
-  build-lint-push-protos:
+  changes:
     runs-on: ubuntu-latest
-    environment: BUF
+    permissions:
+      pull-requests: read
+    outputs:
+      proto: ${{ steps.filters.outputs.proto }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: dorny/paths-filter@v2
+        id: filters
+        with:
+          filters: |
+            proto:
+              - 'crates/astria-proto/**'
+              - '.github/workflows/proto.yml'
```

Comment:
what does this job do? it doesn't look like it runs anything?

Reply:
It provides information about whether or not there were changes to the proto directory or build steps. The output then allows other jobs to run or be skipped based on the outcome.

I think having a seperate job for this is better than a step so it doesn't look like "build" passed when nothing was built.

Reply:
I see, should the `if: ${{ needs.changes.outputs.proto == 'true' }}` be added to the `push` job also?

Reply:
Nope, it requires the build & lint to run